chore: reworking nix module

Mikilio opened the pull request:
Added a systemd service as it allows new configurations to apply on home-manager profile activation.
Also added settings option allowing to configure any option using json. By convention because the config is modifiable by the program itself, I added it to XDG_DATA_HOME by default.

During debugging, I also fixed some minor issues on the side.

More PRs to come

- [x] All settings configurable via nix
- [ ] VPN settings
- [ ] Introducing settings to make non-essential features optional
- [ ] OSD
- [ ] make specific features optional (to tailor to minimalists or people that prefer other tools for specific use-cases)
- [ ] setup devenv with that allows for linting with minimal warnings or errors

Do you maintain a todo list of any kind? If so, it would be nice to make it public in some way, so I don't work on features you're already working on.

Mikilio committed:
chore: reworking nix module

Added a systemd service as it allows new configurations to apply on
home-manager profile activation.

Mikilio committed:
feat: added nix settings and made run optional

This is the promised settings.
Run was made optional by allowing to provide a command to execute
instead. This is because I like to use anyrun more, so I wanted to be
able to use it. The previous behavior is still the default.

!! This changes the default location for the settings to data_home

TheRiceCold reviewed (changes requested):
I think this should be an empty string instead null
![2024-07-15_18-05-05](https://github.com/user-attachments/assets/85797877-fdb7-43f0-9cd9-94f3291f05dc)

Also change this variable to a const to fix the lint error.
I suggest you install `bun` or something similar(npm, yarn, pnpm, etc) so you could check the lints before you push the code. 
![image](https://github.com/user-attachments/assets/04e089eb-3aff-4a7f-a796-e9ee7218eb23)

Mikilio commented:
I see, thank you! Actually I had red warnings all over so I was wondering what environment you were using. Can I add a devenv for the project?  Also: what about ai.proxyUrl? Do you want to avoid null in general?

Mikilio committed:
chore: refactor options

This includes some refactoring that I did in order to make the code
less hard for me to debug.

This also fixes two bugs:

1. Configs are now proper json and not '.' delimeted ids with values.
2. Removed critical error occuring in Dashboard regarding applications

Mikilio commented:
The refactor is obviously opinionated, you insist I can make it more similar to your previous layout.

Mikilio committed:
fix: dependency scan

hyprshade was missing in the package wich made me notice that the
dependency detection function was bugging.
This fixes the bug and also adds hyprshade to the nix package
